### PR TITLE
test: for upgrade compatibility tests retain assigned container ip addresses on upgrade

### PR DESCRIPTION
**Pull request by rboyer**
### Description

For tests that stop and start or upgrade the consul container this ensures that the original networking information is preserved and likely improves the reliability of a quick serf convergence and cuts down on flakiness.

Use a synthetic pod construct to hold onto the IP address in the interim.

### CODE REVIEW

**Review by eculver — APPROVED** [review]
LGTM.

File: test/integration/consul-container/libs/node/consul-container.go [hunk 1]
```diff
@@ -112,25 +117,37 @@ func NewConsulContainer(ctx context.Context, config Config) (Node, error) {
...
 		ctx:       ctx,
-		req:       req,
+		podReq:    podReq,
+		consulReq: consulReq,
 		dataDir:   tmpDirData,
 	}, nil
 }
 
-func newContainerRequest(config Config, name, configFile, dataDir, license string) testcontainers.ContainerRequest {
+const pauseImage = "k8s.gcr.io/pause:3.3"
```
Comment (acpana) [inline test/integration/consul-container/libs/node/consul-container.go:132]: i'm a bit biased to put const at the top or bottom of the file for readibility; do you have any preferences?

File: test/integration/consul-container/libs/node/consul-container.go [hunk 2]
```diff
@@ -112,25 +117,37 @@ func NewConsulContainer(ctx context.Context, config Config) (Node, error) {
...
+
+func newContainerRequest(config Config, name, configFile, dataDir, license string) (podRequest, consulRequest testcontainers.ContainerRequest) {
 	skipReaper := isRYUKDisabled()
 
-	return testcontainers.ContainerRequest{
-		Image:        consulImage + ":" + config.Version,
-		ExposedPorts: []string{"8500/tcp"},
-		WaitingFor:   wait.ForLog(bootLogLine).WithStartupTimeout(10 * time.Second),
+	pod := testcontainers.ContainerRequest{
```
Comment (acpana) [inline test/integration/consul-container/libs/node/consul-container.go:137]: hm, there could be a `newPod` func that wraps around to create a `pod`. Same for `app`. That way we could mix and match down the line.

**Review by acpana — COMMENTED** [review]
small comments, otherwise LGTM

**Pull request merged**